lib/data/multipleBackendGateway.js

```diff
@@ -77,14 +173,28 @@ const multipleBackendGateway = {
                     }
                     cb();
                 });
+            } else if (client.clientType === 'aws_s3') {
+                if (deep) {
+                    awsArray.push(location);
+                    cb();
+                } else {
+                    checkAWSHealth(client, location, cb);
+                }
             } else {
-                // if backend isn't 'scality', it will be 'mem' or 'file',
-                // for which the default response is 200 OK
+                // if backend type isn't 'scality' or 'aws_s3', it will be
+                //  'mem' or 'file', for which the default response is 200 OK
                 multBackendResp[location] = { code: 200, message: 'OK' };
                 cb();
             }
         }, () => {
-            callback(null, multBackendResp);
+            if (awsArray.length > 0) {
+                const randomAWS = awsArray[Math.floor(Math.random() *
+                    awsArray.length)];
+                const checkThisOne = clients[randomAWS];
+                checkAWSHealth(checkThisOne, randomAWS, callback);
```

Comment:
what if there are aws and other backends?  won't you just be returning the aws responses here?

if you don't have a test for healthcheck on multiple backends that involve multiple aws buckets and other backends, please add.